src/plugins/data/public/search/aggs/agg_configs.ts

```diff
@@ -68,21 +73,24 @@ export type IAggConfigs = AggConfigs;
 export class AggConfigs {
   public indexPattern: IndexPattern;
   public timeRange?: TimeRange;
+  private fieldFormats: FieldFormatsStart;
   private readonly typesRegistry: AggTypesRegistryStart;
 
   aggs: IAggConfig[];
 
   constructor(
     indexPattern: IndexPattern,
     configStates: CreateAggConfigParams[] = [],
-    opts: AggConfigsOptions
+    opts: AggConfigsOptions,
+    { fieldFormats }: AggConfigsDependencies
```

Comment:
`AggConfigsOptions` & `AggConfigsDependencies` are essentially the same thing, perhaps `Options` was just a strange choice of name (my bad).

I think these two should be combined into `AggConfigsDependencies`, as both fieldFormats and aggTypes represent stateful services that AggConfigs relies on as a dependency:

```js
new AggConfigs(indexPattern, configStates, { typesRegistry, fieldFormats });
```

Reply:
done